pallets/pallet-bonded-coins/src/curves/lmsr.rs

```diff
@@ -34,13 +34,27 @@ where
 	Parameter: FixedSigned + PartialOrd<Precision> + From<Precision> + ToFixed,
 	<Parameter as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign,
 {
-	fn calculate_exp_term(&self, supply: Parameter, x: Parameter) -> Result<Parameter, ArithmeticError> {
-		supply
-			.checked_sub(x)
-			.ok_or(ArithmeticError::Underflow)?
-			.checked_div(self.m)
-			.ok_or(ArithmeticError::DivisionByZero)
-			.and_then(|exponent| exp::<Parameter, Parameter>(exponent).map_err(|_| ArithmeticError::Overflow))
+	fn lse(&self, supply: Vec<Parameter>) -> Result<Parameter, ArithmeticError> {
+		// Find the maximum value in the supply for numerical stability
+		let max = supply.iter().max().ok_or(ArithmeticError::Underflow)?;
+
+		// Compute the sum of the exponent terms, adjusted by max for stability
+		let e_term_sum = supply.iter().try_fold(Parameter::from_num(0), |acc, x| {
+			let exponent = x
+				.checked_sub(*max)
```

Comment:
How can you substract the max from any other value less than max and not get underflow?

Reply:
The numbers are signed. Only the input from the user is unsinged